src/pool/srv_pool.c

```diff
@@ -1194,9 +1193,6 @@ pool_svc_step_down_cb(struct ds_rsvc *rsvc)
 static void
 pool_svc_drain_cb(struct ds_rsvc *rsvc)
 {
-	struct pool_svc *svc = pool_svc_obj(rsvc);
-
-	ds_rebuild_leader_stop(svc->ps_uuid, -1);
```

Comment:
It does make sense to stop the rebuild leader in drain or step_down callbacks. Why removing the call here?

Reply:
rebuild should not stop during pool leader changes, and we only make master rank to -1,  so the IV related ops will retry with ENOTLEADER. 

And once the pool map/IV is refreshed, then the operation will continue.

Reply:
Could you elaborate why the rebuild leader should not stop when the PS leader steps down?

Reply:
When the leader step down, it will update the master rank to -1 see fini_svc_pool().  Then all retryable IV operations will be in the loop of DER_NOTLEADER see iv_op().

if the new leader step up, and broadcast the pool map, then the new pool map/master rank is being updated on each servers, then all IV operation will continue.  And rebuild does not need stop neither, because the new leader will restart the rebuild process, which send scan rpc to all servers, and each server can handle these "resend" rebuild rpc anyway.

If it is destroy process, then rebuild/iv needs stop, ds_rebuild_abort/iv_ns_stop are only needed in this case.

Reply:
Thanks. It indeed sounds _unnecessary_ to stop rebuild leader activities during PS step down. I still think it would be good to do so, instead of leaving them out there for some new leader or pool destroy. If we move IV retry outside of iv_op to rebuild code, then we should be able to stop rebuild leader activities during PS drain/step down, correct?

Reply:
rebuild restart has a lot overhead, rescan/rebuild duplicate objects etc, besides leader switch does not mean pool service will be stopped, if the rebuild can handle leader switch internally, why would we restart rebuild.

". If we move IV retry outside of iv_op to rebuild code, then we should be able to stop rebuild leader activities during PS drain/step down, correct?"
Sorry, not sure I follow here.  anyway rebuild can handle leader switch for the moment, which is not that complicate, if that is your concern? or I miss the question.